### ISSUE TITLE
Frame.to_npz writes partial data when frame has object dtypes

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#522]
**Description**

Object dtypes are not supported in `to_npz`, but here is an example where `to_npz` fails as expected, but also writes a partial result to its output file. I think static frame should not write anything to the file if `to_npz` fails. 

**Example**

```python
from datetime import date
import static_frame as sf

tgt = '/tmp/frame.npz'
sf.Series([1, 2, 3], name=date.today()).to_frame().to_npz(tgt) # Fails because the sole column has an `object` label.
```

After this, `/tmp/frame.npz` exists

```
cat /tmp/frame.npz
PK!__values_columns_0__.npy⏎  
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
>>> import static_frame as sf
>>> sf.Platform.display()
<Series: platform>
<Index>
platform           Linux-5.15.0-41-generic-x86_64-with
sys.version        3.8.12 (default, Jul  5 2022, 10:53:07) [GCC 7.5.0]
static-frame       0.9.6
numpy              1.17.4
pandas             0.24.2
xlsxwriter         1.1.5
openpyxl           3.0.9
xarray             0.18.0
tables             3.6.1
pyarrow            0.17.0
msgpack            (1, 0, 0)
msgpack_numpy      None
<<U13>             <object>
```

**Comment by flexatone** [comment]
Thanks for isolating this issue. I believe this is essentially the same as https://github.com/InvestmentSystems/static-frame/issues/411. I will try to address this soon.

**Comment by flexatone** [comment]
This has been corrected and will be part of the 0.9.12 release.
